packages/language-server/src/plugins/marko/index.ts

```diff
@@ -0,0 +1,121 @@
+import {
+  LanguageServicePlugin,
+  LanguageServicePluginInstance,
+} from "@volar/language-service";
+import { TextDocument } from "vscode-languageserver-textdocument";
+import { URI } from "vscode-uri";
+
+import { MarkoVirtualCode } from "../../language";
+import { provideCompletions } from "./complete";
+import { provideDefinitions } from "./definition";
+import { provideHover } from "./hover";
+import { provideValidations } from "./validate";
+// import { provideDocumentSymbols } from "./document-symbols";
+
+export const create = (
+  _: typeof import("typescript"),
+): LanguageServicePlugin => {
+  return {
+    name: "marko",
+    capabilities: {
+      hoverProvider: true,
+      definitionProvider: true,
+      diagnosticProvider: {
+        interFileDependencies: false,
+        workspaceDiagnostics: false,
+      },
+      completionProvider: {
+        triggerCharacters: [
+          ".",
+          ":",
+          "<",
+          ">",
+          "@",
+          "/",
+          '"',
+          "'",
+          "`",
+          " ",
+          "=",
+          "*",
+          "#",
+          "$",
+          "+",
+          "^",
+          "(",
+          "[",
+          "-",
+        ],
+      },
+    },
+    create(context): LanguageServicePluginInstance {
+      return {
+        // TODO: Is this necessary?
+        // provideDocumentSymbols(document, token) {
+        //   if (token.isCancellationRequested) return;
+        //   return worker(document, (virtualCode) => {
+        //     return provideDocumentSymbols(virtualCode);
+        //   });
+        // },
```

Comment:
I'm probably missing something - but I wasn't sure if Marko specific document symbols were still necessary and that couldn't be provided by TS or HTML.